libenv/sysinfo.c

```diff
@@ -2311,6 +2317,31 @@ static int Linux_Mandriva_Version_Real(EvalContext *ctx, char *filename, char *r
 
 /******************************************************************/
 
+static int Linux_Amazon_Version(EvalContext *ctx)
+{
+    char buffer[CF_BUFSIZE];
+
+    // Amazon Linux AMI release 2016.09
+
+    if (ReadLine("/etc/system-release", buffer, sizeof(buffer)))
+    {
+       if (strstr(buffer, "Amazon"))
+       {
+           char version[CF_MAXVARSIZE], class[CF_MAXVARSIZE];
+           EvalContextClassPutHard(ctx, "amazon", "inventory,attribute_name=none,source=agent");
+           sscanf(buffer, "%*s %*s %*s %*s %s", version);
+           CanonifyNameInPlace(version);
+           snprintf(class, CF_MAXVARSIZE, "amazon_%s", version);
```

Comment:
what about `amazonlinux_...`? (since you are already setting the generic `amazonlinux` class)